### PR TITLE
qemu_vm: Correct numa_nodeid missing issue

### PR DESCRIPTION
**Pull request by vivianQizhu**
id: 1916550
Signed-off-by: Qianqian Zhu <qizhu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
@huangyum Would you help review? Thanks.

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1731,7 +1731,9 @@ def sort_key(dev):
         if devices.has_option('numa hmat-lb,.*'):
```
Comment (huangyum) [inline virttest/qemu_vm.py:1731]: Prefer to fix the issue by not jumping into this if block when no hmat-lb is defined.

Comment (vivianQizhu) [reply virttest/qemu_vm.py:1731]: @huangyum Cause `hmat-lb` is defined based on `guest_numa_nodes`, what do you think about this:
```
        if devices.has_option('numa hmat-lb,.*'):
            for numa_node in params.objects('guest_numa_nodes'):
                numa_params = params.object_params(numa_node)
                if not numa_params.get('numa_hmat_lb'):
                    continue
                nodeid = numa_params['numa_nodeid']
                ...
```

Comment (huangyum) [reply virttest/qemu_vm.py:1731]: I don't have better idea. This looks good to me.

**Comment by vivianQizhu** [comment]
@huangyum Updated, please help check, thanks.

**Comment by MiriamDeng** [comment]
Tried the bug on ppc, the fix worked well. 
...
'11-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_default.prealloc' Finshed 2021-01-20 02:13:38 2021-01-20 02:15:17 PASS 99.338421
'12-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_default.no_prealloc' Finshed 2021-01-20 02:15:17 2021-01-20 02:17:03 PASS 105.860562
'13-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_bind.prealloc' Finshed 2021-01-20 02:17:03 2021-01-20 02:18:42 PASS 98.613583
'14-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_bind.no_prealloc' Finshed 2021-01-20 02:18:42 2021-01-20 02:20:21 PASS 98.914669
'15-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_interleave.prealloc' Finshed 2021-01-20 02:20:21 2021-01-20 02:22:01 PASS 100.515052
'16-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_interleave.no_prealloc' Finshed 2021-01-20 02:22:02 2021-01-20 02:23:46 PASS 104.195126
'17-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_preferred.prealloc' Finshed 2021-01-20 02:23:46 2021-01-20 02:25:26 PASS 100.164822
'18-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_policy.policy_preferred.no_prealloc' Finshed 2021-01-20 02:25:26 2021-01-20 02:27:08 PASS 101.708993
'19-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.numa_memdev_options.numa_mlock.policy_default.prealloc' Finshed 2021-01-20 02:27:08 2021-01-20 02:28:40 PASS 92.014544
'20-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-
...

**Review by huangyum — APPROVED** [review]
ACK.
 (1/1) Host_RHEL.m8.u4.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.numa_hmat.q35: PASS (36.39 s)

**Comment by vivianQizhu** [comment]
Thanks @huangyum @MiriamDeng

**Pull request merged**